.ci/patch_for_twisted_trunk.sh

```diff
@@ -1,8 +1,22 @@
 #!/bin/sh
 
-# replaces the dependency on Twisted in `python_dependencies` with trunk.
+# replaces the dependency on Twisted with Twisted's trunk.
 
-set -e
+set -xe
 cd "$(dirname "$0")"/..
 
-sed -i -e 's#"Twisted.*"#"Twisted @ git+https://github.com/twisted/twisted"#' synapse/python_dependencies.py
+# We could make full use of the poetry machinery (rather than just `pip install ...`)
+# so that we can catch dependency resolver problems that could arise from twisted
+# bumping its dependencies...
+# sed -ibackup -e 's!^Twisted = .*!Twisted = { git = "https://github.com/twisted/twisted.git", rev = "trunk" }!' pyproject.toml
+# poetry lock --no-update
+# poetry install --no-interaction --extras "all test"
+
+# ...except we run into https://github.com/python-poetry/poetry/issues/5311, where
+# poetry insists on installing an old version of treq, which isn't actually compatible
+# with recent twisted releases. So let's just install twisted trunk using pip.
```

Comment:
No; that comment arrived a little after we settled on using pip. (And as Sean writes above, it would only catch problems which arise from Twisted bumping its dependencies, which seems unlikely at this point in Twisted's lifecycle.)